b2g/components/RemoteControlService.jsm

```diff
@@ -158,8 +158,20 @@ this.RemoteControlService = {
   // 0000 and random generated number to a new string
   // Then get last 4 characters as new PIN code
   generatePIN: function() {
+    // Generate a random number in [0, 2^16 - 1]
+    let random = new Uint16Array(1);
+    let crypto = Services.wm.getMostRecentWindow("navigator:browser").crypto;
+
+    // Make sure this._pin exists and it's not zero
+    do {
+      crypto.getRandomValues(random);
+      // Get last needed digits of the random number
+      this._pin = random[0] % Math.pow(10, PINCODE_LENGTH);
+    } while(!this._pin);
```

Comment:
No, a fallback is bad in this case. Simply fail here. If there's no randomness, you must not do anything. Otherwise you cant guarantee any kind of security.
It doesn't really matter what the actual error is (QuotaExceededError is the only one it can officially throw, but it might as well get stuck and not return anything). You'll probably need some kind of error UI to tell the user that something went wrong in this case. I'd suggest to try to get the pin a couple of times and hard fail with user info in case all attempts fail.